code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -36,9 +36,13 @@
 
 
 /obj/structure/closet/Initialize(mapload)
-	..()
 	if(mapload && !opened)		// if closed, any item at the crate's loc is put in the contents
+		. = TRUE	//late initialize so we come back here
+	else if(initialized)

```

Comment:
Mapload initialization is a bit special in that it can return TRUE optionally if it wants to be called again after everything else. https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/atoms.dm#L45